### DISCUSSION TITLE
Security of Login/Consent accept endpoints

### DISCUSSION THREAD

**Discussion by jburns42891** [ory/hydra#3355] [Q&A] [answered]
Hello,

Had a question surrounding the security of Hydra’s `/accept` endpoints and the flow between Hydra and the identity provider in general. To be clear, I'm referring to calls to:

`oauth2/auth/requests/login/accept?login_challenge=12345`
`oauth2/auth/requests/consent/accept?consent_challenge=12345`

 It seems like anyone that had access to the `login_challenge` parameter could theoretically make a `PUT` call to the `/accept` endpoint with _any_ subject to verify it - I assume that I am missing something here and suspect that it has something to do with using the `force_subject_identifier` option but it’s not clear from the documentation how this should be configured exactly.

Playing around locally with Postman we were able to put in any subject and get a `login_verifier` param returned to us and later on get id and access tokens, all without having to AuthN in our identity app.

If anyone could provide some insight here that would be great! Thanks!

**Comment by aeneasr** [comment, marked as answer]
That's why the endpoint needs to be protected! On the [Ory Network](https://console.ory.sh) for example, you need a personal access token to access these endpoints :)

**Reply by jburns42891** [reply]
Got it - that makes sense. We are looking at self hosting Ory for our use case so sounds like we'd need to implement the authorization for these ourselves. Thanks for the reply!
